File: 0013-roman-to-integer/solution.cpp
class Solution {
public:
    int romanToInt(string s) {
        map <string,int> mp;
        vector<string> sym = {"I","V","X","L","C","D","M"};
        vector<int> numl = {1,5,10,50,100,500,1000};
        for (int i=0;i<sym.size();i++){
            mp[sym[i]] = numl[i];
        };         
        int size = s.length();
        int num = 0;
        int init = 0;
        while (size>=0){
            if (init <= mp[string(1,s[size])]){
                init = mp[string(1,s[size])];
                num +=mp[string(1,s[size])];
                size--;
            }else{
                num -=mp[string(1,s[size])];
                size--;
            }
            


        }
        return num;
    };
};

File: 0013-roman-to-integer/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {
            'I':1,
            "V":5,
            "X":10,
            "L":50,
            "C":100,
            "D":500,
            "M":1000
        }
        ans = 0
        prev = 0

        for ch in reversed(s):
            val = roman[ch]
            if val < prev:
                ans -= val
            else:
                ans += val
            prev = val
        return ans
